pennylane_forest/device.py

```diff
@@ -191,17 +189,16 @@ def __init__(self, wires, shots=1000):
         super().__init__(wires, shots)
         self.reset()
```

Comment:
To avoid the CodeFactor warning:
```
Attribute 'prog' defined outside __init__ (attribute-defined-outside-init)
```

We could do:
```suggestion
        self.prog = Program()
        self._state = None
```

Reply:
I think we have to have a valid `reset()` function since it is part of the `QubitDevice` API, so I don't think we can get around this rule. I added a comment to disable it.

Reply:
We can keep the `reset()` method, but I was suggesting using:
```python
self.prog = Program()
self._state = None
```
in the `__init__` function instead of calling `reset()`. This way the `prog` and `_state` attributes are defined in the `__init__` function and then CodeFactor won't complain anymore